src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json

```diff
@@ -127,6 +127,46 @@
             "Shared/LoginDisplay.IndividualB2CAuth.razor",
             "Shared/LoginDisplay.OrganizationalAuth.razor"
           ]
+        },
+        {
+          "condition": "(!GenerateApi)",
+            "exclude": [
+              "Services/DownstreamWebApi.cs",
+              "Pages/CallWebApi.razor"
+            ]
+        },
+        {
+          "condition": "(!GenerateApiOrGraph)",
+          "rename": {
+            "Shared/NavMenu.NoGraphOrApi.razor": "Shared/NavMenu.razor"
```

Comment:
Why aren't NavMenu.CallsMicrosoftGraph.razor and NavMenu.CallsWebApi.razor excluded?

Reply:
I tried, but there the template engine got confused. So, in the end, after a lot of research, I had to split what is common (the rename), and then what is specific to GenerateGraph and GenerateApi ....
It seems like a bug in the template engine, btw ..

Reply:
I'll see why it's being confused since it looks rather wrong if there are extraneous files.

Reply:
It seems to work for me.

Reply:
Oh ok. thanks for fixing, then!